### DISCUSSION TITLE
Auto generate relationship id columns

### DISCUSSION THREAD

**Discussion by ImogenBits** [sqlalchemy/sqlalchemy#9402] [Ideas]
I often write classes that end up looking like this:
```python
class MyTable(Base):
    id: Mapped[ID]
    other: Mapped[OtherTable] = relationship()
    other_id: Mapped[ID] = mapped_colum(ForeignKey("others.id", init=False)
```
i.e. I have to specify the related object twice, but my code never actually touches the `other_id` attribute. I think it'd be cool to have the relationship attribute automatically generate the actual database column if some argument in the relationship is set. So then the code could be shortened to:
```python
class MyTable(Base):
    id: Mapped[ID]
   other: Mapped[OtherTable] = relationship(auto=True)
```
From what I can tell it "should" be able to automatically determine the needed info. The related column is just the primary key of the table in the relationship. And the column can be named some arbitrarily mangled version of the relationship attribute name, since it won't show up in the python model anyways.

I'm not sure if this is feasible at all or even wanted, but in my use case it'd make things a bit more readable.

**Comment by zzzeek** [comment]
we dont support that directly in SQLAlchemy.  I wrote a blog post many (11) years ago illustrating how people could theoretically roll this themselves at https://techspot.zzzeek.org/2011/05/17/magic-a-new-orm/ - that code would need a few updates to work with modern APIs as well as with typing, but anyway, helpers like that are out of scope for SQLAlchemy proper as we have enough to figure out getting mappings to work with explicit definitions.

\I thought perhaps [SQLModel](https://sqlmodel.tiangolo.com/tutorial/relationship-attributes/define-relationships-attributes/) might offer a pattern like this also but it looks like they still have the explicit FK.
